eng/bump-for-prerelease.js

```diff
@@ -0,0 +1,107 @@
+// @ts-check
+
+/**
+ * Script using the list of changes in all packages to figureout the next version prerelease number.
+ */
+
+const { join, resolve } = require("path");
+const fs = require("fs");
+const stripJsonComments = require("./strip-json-comments");
+const { execSync } = require("child_process");
+
+// Constants
+const PRERELEASE_TYPE = "dev";
+
+const root = resolve(join(__dirname), "..");
+const changeDir = join(root, "common", "changes");
+
+const findAllFiles = async (dir) => {
+  const files = [];
+  for (const file of await fs.promises.readdir(dir)) {
+    const path = join(dir, file);
+    const stat = await fs.promises.lstat(path);
+    if (stat.isDirectory()) {
+      files.push(...(await findAllFiles(path)));
+    } else {
+      files.push(path);
+    }
+  }
+  return files;
+};
+
+const readJsonFile = async (filename) => {
```

Comment:
Can't define these as `async` functions?  Not important, just curious

Reply:
not sure what you mean here, those are `async`

Reply:
I'm referring to the other way of defining the function without assigning a lambda:

```
async function readJsonFile(filename) {
```

Reply:
oh, haha yeah I guess I am used to using const instead of the old function keywords. I think this is coming from doing a lot of react where this is the way to go as you often pipe function and they need the `const` way.
```js
const myComponent = observer(props => props.text);
```
so I just endedup using const all the time for consitency

Reply:
Makes sense!

Reply:
now for non react, this isn't as needed so I guess could switch back to using `function`. It is a subject where I think nobody has a definite answer. THere is small techinical difference between both.